### ISSUE TITLE
(cute_c2) c2CapsuletoPolyManifold giving incorrect results on deep intersections

### ISSUE THREAD

**Issue by TylerGlaiel** [RandyGaul/cute_headers#158]
see the gif. intersection depths appear to be correct, but the normal given is incorrect, causing the polygon to resolve in the wrong direction, and not out of the capsule

![capsule](https://user-images.githubusercontent.com/1686664/67538586-58fbcb00-f694-11e9-9ed5-e7f99e4c8dd4.gif)

(capsule is (0,0), (0,3),  .4)
(poly is mouse + (0,1), (-1,1), (1,-1))

**Comment by RandyGaul** [comment]
Interesting, thanks! Will take a look soon.

**Comment by RandyGaul** [comment]
Looking at this one next.

**Comment by RandyGaul** [comment]
So looks like I just completely forgot about the potential separating axes of the capsule's interior segment, and so I had to write code for that.

I'm not happy with the level of code duplication going here between poly to poly and cap to poly... But it's fine for now. I had to inline write a capsule support and a poly support function, also the switch statement can probably be condensed somewhat.

![cap_to_poly_bugs](https://user-images.githubusercontent.com/1919825/71329097-6804cb00-24d5-11ea-9d1f-1e8c8358d005.gif)

Let me know if you find other problems! Just re-open this issue with a gif, if you do.
